renovate.json

```diff
@@ -0,0 +1,75 @@
+{
+  "$schema":"https://docs.renovatebot.com/renovate-schema.json",
+  "baseBranches":[
+     "main",
+     "release-v0.20",
+     "release-v0.19",
+     "release-v0.18",
+     "release-v0.17"
+  ],
+  "constraints":{
+     "go":"1.22"
+  },
+  "prConcurrentLimit":3,
+  "groupName":"all dependencies",
+  "groupSlug":"all",
+  "lockFileMaintenance":{
+     "enabled":false
+  },
+  "labels":[
+     "release-note-none"
+  ],
+  "extends":[
+     ":gitSignOff"
+  ],
+  "packageRules":[
+     {
+        "groupName":"all dependencies",
+        "groupSlug":"all",
+        "matchBaseBranches":[
+           "main"
+        ],
+        "matchPackagePatterns":[
+           "*"
+        ]
+     },
+     {
+        "enabled":false,
+        "matchBaseBranches":[
+           "release-v0.20",
+           "release-v0.19",
+           "release-v0.18",
+           "release-v0.17"
+        ],
+        "matchPackagePatterns":[
+           "*"
+        ]
+     },
+     {
+        "matchPackageNames":[
```

Comment:
According to the docs: https://docs.renovatebot.com/configuration-options/#matchpackagenames
`matchPackageNames` should contain names of packages. Is `golang` a special name for the compiler version?